SlowCheetah.VisualStudio/TransformationPreviewLogger.cs

```diff
@@ -0,0 +1,125 @@
+        {
+            if (errorListProvider == null)
+            {
+                throw new ArgumentNullException(nameof(errorListProvider));
+            }
+
+            if (hierachy == null)
+            {
+                throw new ArgumentNullException(nameof(hierachy));
+            }
+
+            this.errorListProvider = errorListProvider;
+            this.hierachy = hierachy;
+        }
+
+        /// <inheritdoc/>
+        public void LogError(string message, params object[] messageArgs)
+        {
+            this.AddError(TaskErrorCategory.Error, string.Format(message, messageArgs), null, 0, 0);
+        }
+
+        /// <inheritdoc/>
+        public void LogError(string file, int lineNumber, int linePosition, string message, params object[] messageArgs)
+        {
+            this.AddError(TaskErrorCategory.Error, string.Format(message, messageArgs), file, lineNumber, linePosition);
+        }
+
+        /// <inheritdoc/>
+        public void LogErrorFromException(Exception ex)
+        {
+            this.AddError(ex, TaskErrorCategory.Error, null, 0, 0);
+        }
+
+        /// <inheritdoc/>
+        public void LogErrorFromException(Exception ex, string file, int lineNumber, int linePosition)
+        {
+            this.AddError(ex, TaskErrorCategory.Error, file, lineNumber, linePosition);
+        }
+
+        /// <inheritdoc/>
+        public void LogMessage(string message, params object[] messageArgs)
+        {
+            this.AddError(TaskErrorCategory.Message, string.Format(message, messageArgs), null, 0, 0);
+        }
+
+        /// <inheritdoc/>
+        public void LogWarning(string message, params object[] messageArgs)
+        {
+            this.AddError(TaskErrorCategory.Warning, string.Format(message, messageArgs), null, 0, 0);
+        }
+
+        /// <inheritdoc/>
+        public void LogWarning(string file, int lineNumber, int linePosition, string message, params object[] messageArgs)
+        {
+            this.AddError(TaskErrorCategory.Warning, string.Format(message, messageArgs), file, lineNumber, linePosition);
+        }
+
+        private void AddError(TaskErrorCategory errorCategory, string text, string file, int lineNumber, int linePosition)
+        {
+            this.ShowError(new ErrorTask(), errorCategory, text, file, lineNumber, linePosition);
+        }
+
+        private void AddError(Exception ex, TaskErrorCategory errorCategory, string file, int lineNumber, int linePosition)
+        {
+            this.ShowError(new ErrorTask(ex), errorCategory, null, file, lineNumber, linePosition);
+        }
+
+        private void ShowError(ErrorTask newError, TaskErrorCategory errorCategory, string text, string file, int lineNumber, int linePosition)
+        {
+            newError.Category = TaskCategory.Misc;
+            newError.ErrorCategory = errorCategory;
+            if (file != null)
```

Comment:
do you mean `if (text != null)`? Also, why bother with the null check? You can just set these values to `null` or `0` without an issue.